boards/holybro/kakutef7/src/board_config.h

```diff
@@ -0,0 +1,313 @@
+
+#define GPIO_DRDY_OFF_SPI4_DRDY1_ICM20689    _PIN_OFF(GPIO_SPI4_DRDY1_ICM20689)
+
+
+/*
+ *  Define the ability to shut off off the sensor signals
+ *  by changing the signals to inputs
+ */
+
+#define _PIN_OFF(def) (((def) & (GPIO_PORT_MASK | GPIO_PIN_MASK)) | (GPIO_INPUT|GPIO_PULLDOWN|GPIO_SPEED_2MHz))
+
+
+#define GPIO_DRDY_OFF_SPI5_DRDY7_EXTERNAL1   _PIN_OFF(GPIO_SPI5_DRDY7_EXTERNAL1)
+
+
+#define PX4_SPIDEV_MPU              PX4_MK_SPI_SEL(PX4_SPI_BUS_SENSORS,0)
+#define PX4_SPIDEV_ICM_20689        PX4_MK_SPI_SEL(PX4_SPI_BUS_SENSORS,0)
+#define PX4_SENSOR_BUS_CS_GPIO      {GPIO_SPI4_CS1_ICM20689}
+
+/* I2C busses */
+
+#define PX4_I2C_BUS_EXPANSION       1
+#define PX4_I2C_BUS_LED             PX4_I2C_BUS_EXPANSION
+
+#define BOARD_NUMBER_I2C_BUSES      1
+#define BOARD_I2C_BUS_CLOCK_INIT    {100000}
+
+#define PX4_I2C_OBDEV_BMP280        0x76
+
+/*
+ * ADC channels
+ *
+ * These are the channel numbers of the ADCs of the microcontroller that
+ * can be used by the Px4 Firmware in the adc driver
+ */
+
+/* ADC defines to be used in sensors.cpp to read from a particular channel */
+#define ADC1_CH(n)                  (n)
+#define ADC1_GPIO(n)                GPIO_ADC1_IN##n
+
+/* Define GPIO pins used as ADC N.B. Channel numbers must match below */
+#define PX4_ADC_GPIO  \
+	/* PC2 */  ADC1_GPIO(12),  \
+	/* PC3 */  ADC1_GPIO(13),  \
+	/* PC5 */  ADC1_GPIO(15)
+
+/* Define Channel numbers must match above GPIO pin IN(n)*/
+#define ADC_RSSI_IN_CHANNEL                 /* PC5 */  ADC1_CH(15)
+#define ADC_BATTERY_VOLTAGE_CHANNEL         /* PC3 */  ADC1_CH(13)
+#define ADC_BATTERY_CURRENT_CHANNEL         /* PC2 */  ADC1_CH(12)
+
+#define ADC_CHANNELS \
+	((1 << ADC_BATTERY_VOLTAGE_CHANNEL)       | \
+	 (1 << ADC_BATTERY_CURRENT_CHANNEL)       | \
+	 (1 << ADC_RSSI_IN_CHANNEL))
+
+/* Define Battery 1 Voltage Divider and A per V
+ */
+#define BOARD_BATTERY1_V_DIV         (10.9f)
+#define BOARD_BATTERY1_A_PER_V       (17.f)
+
+/* PWM
+ */
+#define GPIO_TIM5_CH4OUT      /* PA3   T5C4   PWM6 */ GPIO_TIM5_CH4OUT_1
+#define GPIO_TIM8_CH4OUT      /* PC9   T8C4   PWM5 */ GPIO_TIM8_CH4OUT_1
+#define GPIO_TIM1_CH2OUT      /* PE11  T1C2   PWM4 */ GPIO_TIM1_CH2OUT_2
+#define GPIO_TIM1_CH1OUT      /* PE9   T1C1   PWM3 */ GPIO_TIM1_CH1OUT_2
+#define GPIO_TIM3_CH4OUT      /* PB1   T3C4   PWM2 */ GPIO_TIM3_CH4OUT_1
+#define GPIO_TIM3_CH3OUT      /* PB0   T3C3   PWM1 */ GPIO_TIM3_CH3OUT_1
+
+#define DIRECT_PWM_OUTPUT_CHANNELS  6
+
+#define GPIO_TIM5_CH4IN       /* PA3   T5C4   PWM6 */ GPIO_TIM5_CH4IN_1
+#define GPIO_TIM8_CH4IN       /* PC9   T8C4   PWM5 */ GPIO_TIM8_CH4IN_1
+#define GPIO_TIM1_CH2IN       /* PE11  T1C2   PWM4 */ GPIO_TIM1_CH2IN_2
+#define GPIO_TIM1_CH1IN       /* PE9   T1C1   PWM3 */ GPIO_TIM1_CH1IN_2
+#define GPIO_TIM3_CH4IN       /* PB1   T3C4   PWM2 */ GPIO_TIM3_CH4IN_1
+#define GPIO_TIM3_CH3IN       /* PB0   T3C3   PWM1 */ GPIO_TIM3_CH3IN_1
+
+#define DIRECT_INPUT_TIMER_CHANNELS  6
+
+#define BOARD_HAS_LED_PWM              1
```

Comment:
Remove this This is for RGB led to do a cheap cost super bright RGB replacment for I2C ones.

Reply:
Removed.